### ISSUE TITLE
[BUG] Add column to Dataframe is slow

### ISSUE THREAD

**Issue by ayushdg** [rapidsai/cudf#1348]
**Description**
Add column to a Dataframe is slow. Works faster if the data is wrapped as a series before adding to Dataframe

**Steps/Code to reproduce bug**
```
import cudf

n_elem = 100_000_000
list1 = [i for i in range(n_elem)]
```
```
df = cudf.DataFrame()
%time df['1'] = list1

Output:
CPU times: user 12.4 s, sys: 2.36 s, total: 14.7 s
Wall time: 14.7 s
```
vs
```
df = cudf.DataFrame()
%time df['1'] = cudf.Series(list1)

Output:
CPU times: user 4.16 s, sys: 768 ms, total: 4.92 s
Wall time: 4.91 s
```

**Environment details:**
Environment location: Jupyter from docker (nightly image)
Cudf version: '0.6.1+1.g9ca9325'
 

**Additional context**

I have a feeling that creating a Cudf series thrice, one in each function call to `_sanitize_columns` and `_sanitize_values` and one in L789 might translate to the first method taking roughly 3x the time of the second one. What if we do the series creation before the function calls. Thoughts?? https://github.com/rapidsai/cudf/blob/58316cb0c1303253f254b7247402ec0ed9bf357d/python/cudf/dataframe/dataframe.py#L785-L789

**Comment by VibhuJawa** [comment]
Facing a similar slowdown with string list and nvstrings. 

#### Reproduction Code
```python
import cudf
import time
import nvstrings
input_lines = ['Test--String']*50_000_000

## Load strs to cudf directly
df = cudf.DataFrame()
st = time.time()
df['text']= input_lines
et = time.time()
print("Time taken to read directly = {}".format(et-st))
del df

## Load strs to nvstrings and then to divice
st = time.time()
df = cudf.DataFrame()
df['text'] = nvstrings.to_device(input_lines)
et = time.time()
print("Time taken to read via nvstrings = {}".format(et-st))
del df
```

Output 
```
Time taken to read directly = 16.476869821548462
Time taken to read via nvstrings = 3.1028056144714355 
```

Also, I am seeing memory spikes  when creating a column from cudf string where temporary the memory being used is more than the total memory. 

The y  axis is memory in mib and x axis is seconds

![Memory Graph](https://user-images.githubusercontent.com/4837571/55917996-56c59980-5ba6-11e9-9215-1ea6f49301e5.png)

**Comment by ayushdg** [comment]
Should be resolved with #1421
